test/libsolidity/semanticTests/builtins/balance_other_contract.sol

```diff
@@ -0,0 +1,23 @@
+contract Other {
+    constructor() payable {
+    }
+    function getAddress() public returns (address) {
+        return address(this);
+    }
+}
+contract ClientReceipt {
+    Other other;
+    constructor() payable {
+        other = new Other{value:500}();
+    }
+    function getAddress() public returns (address) {
+        return other.getAddress();
+    }
+}
+// ====
+// compileViaYul: also
+// ----
+// constructor(), 2000 wei ->
+// contract.balance -> 1500
+// getAddress() -> 0xf01f7809444bd9a93a854361c6fae3f23d9e23db
+// contract.balance: 0xf01f7809444bd9a93a854361c6fae3f23d9e23db -> 500
```

Comment:
Hmm, looking at this, I think actually we should have `account.balance: <addr>` and `contract.balance` if possible.

Reply:
hmm.. true! it is correct that `account.balance` and `contract.balance` are just synonyms?

Reply:
I would say `account.balance` requires an argument, whereas `contract.balance` cannot accept an argument. The latter always points to the deployed contract in the testing context.